GVFS/GVFS.Notifications/VFSForGit.Mac/StatusMenuItem/Utilities/ProcessRunner.m

```diff
@@ -0,0 +1,59 @@
+#import "ProcessRunner.h"
+
+@interface ProcessRunner()
+
+@property (strong) ProcessFactory processFactory;
+
+@end
+
+@implementation ProcessRunner
+
+- (id) initWithProcessFactory:(ProcessFactory) processFactory
+{
+    if (self = [super init])
+    {
+        self.processFactory = processFactory;
+    }
+    
+    return self;
+}
+
+- (BOOL) tryRunExecutable:(NSURL *) path
+    args:(NSArray *) args
+    output:(NSString **) output
+    error:(NSError **) error
+{
+    NSTask *task = self.processFactory();
+    NSPipe *taskOut = [NSPipe pipe];
+    
+    task.executableURL = path;
+    task.arguments = args;
+    task.standardOutput = taskOut;
+    task.standardError = taskOut;
+    
+    int exitCode = -1;
+    
+    if ([task launchAndReturnError:error])
+    {
+        [task waitUntilExit];
+        
+        exitCode = [task terminationStatus];
+        
+        *output = [[NSString alloc]
```

Comment:
Check output != nil first, otherwise this could be a dereferencing null exception.

Reply:
Added `nullability` annotations, this does compile time null checks. Also added run time parameter asserts to raise just in case we get `null` at run time.